soc/arm/st_stm32/stm32l4/power.c

```diff
@@ -30,7 +30,11 @@ void sys_set_power_state(enum power_states state)
 		LL_DBGMCU_EnableDBGStopMode();
 #endif /* CONFIG_DEBUG */
 		/* ensure HSI is the wake-up system clock */
+#if defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI)
+		LL_RCC_SetClkAfterWakeFromStop(LL_RCC_STOP_WAKEUPCLOCK_MSI);
+#else
 		LL_RCC_SetClkAfterWakeFromStop(LL_RCC_STOP_WAKEUPCLOCK_HSI);
+#endif
```

Comment:
Rather than adding #ifdefery allover, use of a static variable would help readability

Reply:
In our use case it sounds natural to use MSI as wakeup clock when MSI is configured as system clock. This impacts only the programming of bit STOPWUCK in RCC_CFGR.
I'm not aware of other uses cases that would need a variable, how should we manage it?

Reply:
@giansta Something like follows would make the code more readable:
```
#if defined(CONFIG_CLOCK_STM32_SYSCLK_SRC_MSI)
#define STOP_WAKEUP_CLK LL_RCC_STOP_WAKEUPCLOCK_MSI
#else
#define STOP_WAKEUP_CLK LL_RCC_STOP_WAKEUPCLOCK_HSI
#endif
[...]
       LL_RCC_SetClkAfterWakeFromStop(STOP_WAKEUP_CLK);
[...]
```
This would avoid redundant ifdefery.

Reply:
Ah, I understood you were requesting a more structural change using variables. Totally agree on this, thanks. Just rebased.